### ISSUE TITLE
Evaluate any String Match to evaluate as a Quasi Quote / code fragment

### ISSUE THREAD

**Issue by sirinath** [sirthias/parboiled2#126]
I want to treat some matched string values as QQs. Is it possible to add this functionality to Parboiled2?

E,g.
 - `~ eval(_)` and / or
 - `~ qq(_)`, `~ qt(_)`, `~ qp(_)`, `~ cq(_)`, `~ fq(_)`

**Comment by sirinath** [comment]
May be initially as a stating point being able to parse Scala types and materialise them and a way to create new types based on what is parse.

**Comment by sirthias** [comment]
Since you run your parser at application runtime you'd have to carry along the compiler to have it too run at runtime. Possible, but not something that we'd be adding to parboiled2 any time soon I'd say.

**Comment by sirinath** [comment]
In which case is it possible to give the functionality where you can provide some rules (file, input stream, string) at runtime and regenerate the parser?

**Comment by sirthias** [comment]
Not sure I understand what exactly you are asking.
Can you rephrase your question?

**Comment by sirinath** [comment]
At compile time of the application I do not know the full grammar I am parsing. So I want to inject rules and actions through configuration.

So the from your side parser generation is delayed until runtime when the full configuration is given.

**Comment by sirthias** [comment]
Parser generation is macro-based and therefore runs compile-time, not runtime.
If you wanted to generate a parser at runtime you need to carry along the whole compiler.
